package com.github.kurlymarketclone.repository.userDetail;

import com.github.kurlymarketclone.repository.user.User;
import java.util.Map;
import lombok.*;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;
import java.util.List;
import org.springframework.security.oauth2.core.user.OAuth2User;


@Builder
@Getter
@Setter
@AllArgsConstructor
@ToString
public class CustomOauth2UserDetails implements UserDetails, OAuth2User {
    private String myId;
    private String password;
    private List<String> authorities;
    private Integer userId;

    private final User user;
    private Map<String, Object> attributes;

    public CustomOauth2UserDetails(User user, Map<String, Object> attributes) {
        this.user = user;
        this.attributes = attributes;
    }

    @Override
    public Map<String, Object> getAttributes() {
        return attributes;
    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return null;
    }

    @Override
    public String getPassword() {
        return password;
    }

    @Override
    public String getUsername() {
        return myId;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }


}
